include/aws/http/private/h2_decoder.h

```diff
@@ -34,17 +34,26 @@
 
 struct aws_h2_decoder_vtable {
     /* For HEADERS header-block: _begin() is called, then 0+ _i() calls, then _end().
-     * No other decoder callbacks will occur in this time. */
+     * No other decoder callbacks will occur in this time.
+     * If something is malformed, no further _i() calls occur, and it is reported in _end() */
     int (*on_headers_begin)(uint32_t stream_id, void *userdata);
-    int (*on_headers_i)(uint32_t stream_id, const struct aws_http_header *header, void *userdata);
-    int (*on_headers_end)(uint32_t stream_id, void *userdata);
+    int (*on_headers_i)(
```

Comment:
I must have missed this, but why isn't this just ```on_header()```? Likewise with ```on_push_promise()```?

Reply:
I dunno, wanted the trio of callbacks to look like a for-loop
It's not a public API